Branch-SDK/src/main/java/io/branch/referral/util/BranchEvent.java

```diff
@@ -256,6 +269,175 @@ public boolean logEvent(Context context) {
+                                    .addCustomDataProperty("purchase_token", purchase.getPurchaseToken())
+                                    .addContentItems(buos)
+                                    .logEvent(context);
+                        } else {
+                            Log.e("BranchEvent", "Failed to query product details. Error code: " + billingResult.getResponseCode());
+                        }
+                    });
+
+
+                    billingClient.queryProductDetailsAsync(queryProductDetailsParams, (billingQueryResult, productDetailsList) -> {
+                        if (billingQueryResult.getResponseCode() == BillingClient.BillingResponseCode.OK) {
+                            List<BranchUniversalObject> buos = new ArrayList<>();
+                            CurrencyType currency = null;
+                            double revenue = 0.00;
+                            String purchasedProductType = "";
+
+                            for (ProductDetails product : productDetailsList) {
+                                double price;
+                                int quantity;
+
+                                if (product.getProductType() == "inapp_subscription") {
+                                    purchasedProductType = "Subscription";
+                                    currency = CurrencyType.valueOf(product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceCurrencyCode());
+                                    price = product.getSubscriptionOfferDetails().get(0).getPricingPhases().getPricingPhaseList().get(0).getPriceAmountMicros() / 1_000_000;
+                                    quantity = 1;
+                                    revenue += price * quantity;
+                                } else {
+                                    purchasedProductType = "IAP";
+                                    currency = CurrencyType.valueOf(product.getOneTimePurchaseOfferDetails().getPriceCurrencyCode());
+                                    price = product.getOneTimePurchaseOfferDetails().getPriceAmountMicros() / 1_000_000;
+                                    quantity = purchase.getQuantity();
+                                    revenue += price * quantity;
+                                }
+
+                                BranchUniversalObject buo = new BranchUniversalObject()
+                                        .setCanonicalIdentifier(product.getProductId())
+                                        .setTitle(product.getTitle())
+                                        .setContentMetadata(
+                                                new ContentMetadata()
+                                                        .addCustomMetadata("product_type", product.getProductType())
+                                                        .setPrice(price, currency)
```

Comment:
Ah but set price requires a `double` here. In that case you can assign 

`price = product.getOneTimePurchaseOfferDetails().getPriceAmountMicros() / 1_000_000.0;`